src/stream-tcp-inline.c

```diff
@@ -41,9 +41,14 @@ extern int stream_inline;
  *  \retval 0 no
  *  \retval 1 yes
  */
-int StreamTcpInlineMode(void)
+int StreamTcpInlineMode(const Packet *p)
 {
-    return stream_inline;
+    if (stream_inline == PKT_MODE_IDS)
+        return 0;
+    else if (stream_inline == PKT_MODE_IPS)
+        return 1;
+    else /* implied AUTO */
+        return (stream_inline && PacketModeIsIPS(p));
```

Comment:
On 09-09-16 16:49, Giuseppe Longo wrote:
> In src/stream-tcp-inline.c
> <https://github.com/inliniac/suricata/pull/2232#discussion_r78192965>:
> 
>>  {
>> -    return stream_inline;
>> +    if (stream_inline == PKT_MODE_IDS)
>> +        return 0;
>> +    else if (stream_inline == PKT_MODE_IPS)
>> +        return 1;
>> +    else /* implied AUTO */
>> +        return (stream_inline && PacketModeIsIPS(p));
> 
> if PacketModeIsIPS(p) returns 0, the condition is not true, I'm not
> getting the point maybe.

My point is only about stream_inline part of return (stream_inline &&
PacketModeIsIPS(p));. That part is not adding anything.